core/vm/evm.go

```diff
@@ -464,7 +463,7 @@ func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gas uint64,
 	// We add this to the access list _before_ taking a snapshot. Even if the creation fails,
 	// the access-list change should not be rolled back
 	if evm.chainRules.IsBerlin {
-		evm.StateDB.AddAddressToAccessList(address)
+		evm.StateDB.AddAddressToAccessList(address, state.ALVersion|state.ALCodeHash|state.ALNonce, state.AccessListWrite)
```

Comment:
Note to self: this should be ineffective because of what happens on line 444+, except for the code hash which I believe is charged too early (but the failed contract creation work, so even then it should be fine).